src/ClassRegistrator_impl.hpp

```diff
@@ -91,12 +92,14 @@ namespace { // Implementation detail, so it's in an anon namespace
     };
 }
 
-template<typename Class>
-Chibi::ClassRegistrator<Class>::ClassRegistrator(Chibi &chibi, std::string &name) : chibi(chibi), class_name(name) { }
+template <typename Class>
+Chibi::ClassRegistrator<Class>::ClassRegistrator(Chibi &chibi, const std::string &name)
+    : chibi(chibi)
+    , class_name(name) { }
 
-template<typename Class>
-template<typename Return, typename... Args>
-Chibi::ClassRegistrator<Class> &Chibi::ClassRegistrator<Class>::register_method(std::string &&name, Return (Class::*member_function)(Args...)) {
+template <typename Class>
+template <typename Return, typename... Args>
+Chibi::ClassRegistrator<Class> &Chibi::ClassRegistrator<Class>::register_method(const std::string &&name, Return (Class::*member_function)(Args...)) {
```

Comment:
const rvalue ref.
Also why it's rvalue? I don't see any moving inside this method.

Reply:
Afaik I did this to be able to pass char* and strings all the same.